modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloVelocityMojo.java

```diff
@@ -0,0 +1,86 @@
+package org.codehaus.modello.maven;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.modello.plugin.velocity.VelocityGenerator;
+
+/**
+ * Creates files from the model using Velocity templates.
+ *
+ * @author <a href="mailto:brett@codehaus.org">Brett Porter</a>
+ */
+@Mojo( name = "velocity", defaultPhase = LifecyclePhase.GENERATE_SOURCES, threadSafe = true )
+public class ModelloVelocityMojo
+        extends AbstractModelloGeneratorMojo
```

Comment:
should not be a Mojo (ie a Maven plugin) but a Generator (ie a Modello plugin) like every other Modello plugin in Modello: https://github.com/codehaus-plexus/modello/tree/master/modello-plugins

Reply:
Am I missing something ? Each generator has a corresponding mojo in the `maven-modello-plugin`.
The generator is
  https://github.com/codehaus-plexus/modello/tree/40e30c867e48903883c094a7618082a737d75d51/modello-plugins/modello-plugin-velocity
and the mojo is
  https://github.com/codehaus-plexus/modello/blob/40e30c867e48903883c094a7618082a737d75d51/modello-maven-plugin/src/main/java/org/codehaus/modello/maven/ModelloVelocityMojo.java
That seems to be how all generators are written.

Reply:
sorry, it's me who overlooked that this Mojo was perfectly put in the modello-maven-plugin...
sorry for the noise